realm/src/main/java/io/realm/Realm.java

```diff
@@ -1403,6 +1694,73 @@ public void executeTransaction(Transaction transaction) {
     }
 
     /**
+     * Similar to {@link #executeTransaction(Transaction)} but runs asynchronously from a worker thread
+     * @param transaction {@link io.realm.Realm.Transaction} to execute.
+     * @param callback optional, to receive the result of this query
+     * @return A {@link io.realm.Realm.Request} representing a cancellable task
+     */
+    public Request executeTransaction(final Transaction transaction, final Transaction.Callback callback) {
+        if (transaction == null)
+            throw new IllegalArgumentException("transaction should not be null");
+
+        if (callback == null)
+            throw new IllegalArgumentException("callback should not be null");
+
+        // will use the Looper of the caller thread to post the result
+        final Handler handler = new Handler();
+
+        // We need to use the same configuration to open a background SharedGroup (i.e Realm)
+        // to perform the transaction
+        final RealmConfiguration realmConfiguration = getConfiguration();
+
+        final Future<?> pendingQuery = asyncQueryExecutor.submit(new Runnable() {
```

Comment:
not sure of your point, calling `pendingQuery.cancel` will change the state of the FutureTask to cancel (if it's still in the working Queue) if the Executors already started the task it's too late, the best effort is to keep checking `interrupt` within the runnable